src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -374,6 +374,24 @@ public static void Sleep(int millisecondsTimeout)
         internal static ulong CurrentOSThreadId => GetCurrentOSThreadId();
 #endif
 
+        internal static bool ShouldSpinWait { get; } = DetermineShouldSpinWait();
+
+        // In cases where the cpu is limited due to cpu quota limits or there's only a single processor
+        // choosing not to spin wait might allow for more actual work to be done.
+        // However it's possible to spin wait in those scenarios by enabling the SpinWaitWhenCpuQuotaIsLimited config setting.
+        private static bool DetermineShouldSpinWait()
+        {
+            return
+                !Environment.IsSingleProcessor &&
+                (
+                    !Environment.IsCpuQuotaLimited ||
+                    AppContextConfigHelper.GetBooleanConfig(
+                        "System.Threading.ThreadPool.SpinWaitWhenCpuQuotaIsLimited",
+                        "DOTNET_ThreadPool_SpinWaitWhenCpuQuotaIsLimited",
```

Comment:
Maybe these should be under `Threading` instead of `ThreadPool`, as in:
```suggestion
                        "System.Threading.SpinWaitWhenCpuQuotaIsLimited",
                        "DOTNET_Threading_SpinWaitWhenCpuQuotaIsLimited",
```

Also, maybe it would be better to suggest "reducing" spin-waiting rather than to suggest it would be disabled, to leave room for that ([comment](https://github.com/dotnet/runtime/pull/113184#discussion_r1992493706)). Maybe something like:

```suggestion
                        "System.Threading.ReduceSpinWaitingWhenCpuQuotaIsLimited",
                        "DOTNET_Threading_ReduceSpinWaitingWhenCpuQuotaIsLimited",
```

With that, perhaps the property on the `Thread` class could be removed in favor of each component doing whatever is appropriate based on the property on `Environment`. Could still start with disabling spin-waiting by default.